skbio/metadata/_interval.py

```diff
@@ -0,0 +1,699 @@
+
+    def sort(self, ascending=True):
+        '''Sort interval features by their coordinates.
+
+        It sorts by the start coordinate first. If they are the same between
+        two interval features, they will be sorted by comparing their end
+        coordinates. For example, an interval feature with [(1, 2), (4, 7)]
+        will be sorted in front of another one with [(1, 2), (3, 8)].
+
+        Parameters
+        ----------
+        ascending : bool
+            sort in ascending or descending coordinates.'''
+        self._intervals = sorted(
+            self._intervals,
+            key=lambda i: [i.locations[0][0], i.locations[-1][1]],
+            reverse=not ascending)
+
+    def add(self, locations, boundaries=None, metadata=None):
+        """Add a feature to the metadata object.
+
+        This method creates an ``Interval`` object and insert it into
+        the ``IntervalMetadata`` object.
+
+        Parameters
+        ----------
+        locations : iterable of tuple of ints
+            A list of locations associated with the Interval object.
+        boundaries : iterable of tuple of bool
+            A list of boundaries associated with the locations.
+        metadata : dict
+            A dictionary of key word attributes associated with the
+            Interval object.
+
+        Returns
+        -------
+        Interval
+            The ``Interval`` object added.
+
+        See Also
+        --------
+        skbio.metadata.Interval._add
+        """
+        # Add an interval to the tree. Note that the add functionality is
+        # built within the Interval constructor.
+        return Interval(interval_metadata=self,
+                        locations=locations,
+                        boundaries=boundaries,
+                        metadata=metadata)
+
+    @_rebuild_tree
+    def _query_interval(self, location):
+        """Yield ``Interval`` objects that overlap with the location."""
+        _assert_valid_location(location)
+
+        start, end = location
+        intvls = self._interval_tree.find(start, end)
+        # if a ``Interval`` has many non-contiguous spans and
+        # multiple of them overlap with the location, then
+        # this ``Interval`` object will be returned
+        # multiple times. So we need to remove duplicates.
+        seen = set()
+        for intvl in intvls:
+            if id(intvl) not in seen:
+                seen.add(id(intvl))
+                yield intvl
+
+    def _query_attribute(self, metadata, intervals=None):
+        """Yield ``Interval`` objects based on query attributes.
+
+        Parameters
+        ----------
+        metadata : dict or ``None``
+            If it is ``None``, return empty iterator; if it is
+            ``{}``, return an interator of all the ``Interval``
+            objects.
+        intervals : an iterable of ``Interval`` objects
+        """
+        if metadata is None:
+            return
+            yield
```

Comment:
This yield is unnecessary as there is still one at the bottom (so Python knows its a generator). Plus it is totally bombing your test coverage (1 line missing!) :smile: